tests/expecter.go

```diff
@@ -228,7 +229,16 @@ func ExpectBatchWithValidatedSend(expecter expect.Expecter, batch []expect.Batch
 
 			// Remove the \n since it is translated by the console to \r\n.
 			previousSend = strings.TrimSuffix(previousSend, "\n")
-			bExp.R = fmt.Sprintf("%s%s%s", previousSend, "((?s).*)", bExp.R)
+			// Somtimes after logging in, the previous command will be removed from the buffer,
+			// for that we allow it also to match without the previous command, but only if it is at
+			// the beggining of the line.
+			// Plus in some consoles the prompt is read with a preceded white space, and sometimes it doesn't, so we
+			// allow that as well.
+			bExp.R = fmt.Sprintf("(%s%s%s"+`|^\s?`+"%s*%s)", previousSend, "((?s).*)", bExp.R, CRLF, bExp.R)
```

Comment:
This expression breaks the whole idea of `ExpectBatchWithValidatedSend`.
The idea is to make sure the expected expression is the result of the sent command and not a leftover of the buffer.
With your new addition to the expression we rollback to the previous situation where we could find the expected expression somewhere in the buffer left over.

Reply:
I am not sure it completely breaks it. it may weakens it a bit, but we gained using it also in login batches. the point is to avoid leftovers from previous commands, if we are matching the expression only if is at the beginning of the buffer (with `^` ) we should almost completely avoid matching leftovers, don't we?

I thought of scenario were what I suggested might fail but I think it is really rare and perhaps worth the risk: 
```
					&expect.BSnd{S: "sudo su -\n"},
					&expect.BExp{R: tests.PromptExpression},
					&expect.BSnd{S: "mount /dev/sda /mnt\n"},
					&expect.BExp{R: tests.PromptExpression},
```
Here for example  after we send `sudo su \n` we expect a prompt, and on the second time we expect a prompt we might, somehow, match the first prompt, but it will happen only if the first prompt is appearing right at the beginning of the buffer with nothing besides white-spaces, `\r` and `\n` preceding it. 
if, for example, `sudo su \n` will precede it, the right side of the `|` won't match, and we are back at our safe zone.

make sense, or I'm talking BS?

Reply:
yes, the buffer leftover may start with the expected result (we have pretty high possibly for it to happen when the expected result if the prompt).
Anyway, the login method start with sending `\n` and then expecting for an expression that is not the result of `\n`. Therefore, those methods are not suitable to be used by the `ExpectBatchWithValidatedSend`.

We have only 3 login method. And you use `ExpectBatchWithValidatedSend` for only two of them. It doesn't worth adapting `ExpectBatchWithValidatedSend` just for those 2 methods. We should make sure those ,methods are safe without `ExpectBatchWithValidatedSend`.

Reply:
OK then. I will drop the second commit